validator/validators/performance/performance_validator.go

```diff
@@ -42,6 +61,20 @@ func (s *PerformanceValidator) GenerateLoad() (err error) {
 }
 
 func (s *PerformanceValidator) CheckData(startTime, endTime time.Time) error {
```

Comment:
nit: The name `CheckData()` does not match with what this function is doing. Fetching metrics, post-processing them and uploading a report to a database...

I realize the name is part of the interface.
Maybe add another optional function to the interface for `PublishResults()`.

Or maybe that causes a different problem - having big interfaces that are loosely implemented.
Don't want an "Animal" interface with functions like "Speak()" since not animals can speak ;)

Reply:
Agree. However, an interface which has general method would be a better case here since not Animal can all `Fly()` but most of all the `Animals` can `Speak`.  The plan for me would be replace all our existing test with the validator and `CheckData` can be applied to all our existing tests !!